app/src/main/java/com/supertokens/session/SuperTokens.java

```diff
@@ -102,8 +104,27 @@ static String getApiDomain(@NonNull String url) throws MalformedURLException {
 
     @SuppressWarnings("unused")
     public static boolean doesSessionExist(Context context) {
-        String idRefreshToken = IdRefreshToken.getToken(context);
-        return idRefreshToken != null;
+        try {
+            JSONObject frontToken = FrontToken.getToken(context);
+
+            if (frontToken == null) {
+                return false;
+            }
+
+            long currentTimeInMillis = System.currentTimeMillis();
+
+            long accessTokenExpiry = frontToken.getLong("ate");
+
+            if (accessTokenExpiry < System.currentTimeMillis()) {
+                Utils.LocalSessionState localSessionState = Utils.getLocalSessionState(context);
+                Utils.Unauthorised response = SuperTokensHttpURLConnection.onUnauthorisedResponse(localSessionState, context);
+                return response.status == Utils.Unauthorised.UnauthorisedStatus.RETRY;
+            }
+        } catch (Exception e) {
+            return false;
+        }
```

Comment:
Catching all exceptions is a bad idea in general. We only want to return false in the case the `onUnauthorisedResponse` throws an error right? So maybe keep the catch clause specific to those exceptions